File: app/src/main/java/com/dof100/morsenotifier/ActivityReminders.java
package com.dof100.morsenotifier;

import android.app.Activity;
import android.app.AlertDialog.Builder;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.ListView;
import android.widget.AdapterView.OnItemClickListener;

public class ActivityReminders extends Activity implements OnClickListener, MyRemindersArrayAdapter.a {
   private MyReminders a;
   private MyRemindersArrayAdapter b;
   private ListView c;

   public void a(final MyReminder var1, int var2, View var3) {
      MyLog.log("ActivityReminders.onRowButtonClick");
      if (var3.getId() == R.id.b_delete) {
         MyLog.log("ActivityReminders.onRowButtonClick b_delete");
         Builder var5 = new Builder(this);
         var5.setTitle(R.string.uninstall_free_title);
         var5.setMessage(R.string.title_activity_advanced);
         var5.setPositiveButton(R.string.action_yes, new android.content.DialogInterface.OnClickListener() {
            public void onClick(DialogInterface var1x, int var2) {
               ActivityReminders.this.b.remove(var1);
               ActivityReminders.this.a.b(ActivityReminders.this);
               ActivityReminders.this.b.notifyDataSetChanged();
               MyJob.scheduleNextReminder((Context)ActivityReminders.this);
               ActivityReminders.this.c.invalidate();
               var1x.dismiss();
            }
         });
         var5.setNegativeButton(R.string.action_no, new android.content.DialogInterface.OnClickListener() {
            public void onClick(DialogInterface var1, int var2) {
               var1.dismiss();
            }
         });
         var5.create().show();
      } else {
         if (var3.getId() == R.id.b_edit) {
            MyLog.log("ActivityReminders.onRowButtonClick b_edit");
            Intent var4 = new Intent(this, ActivityReminder.class);
            var4.putExtra("OBJECTINDEX", var2);
            this.startActivityForResult(var4, 1);
         }

      }
   }

   protected void onActivityResult(int var1, int var2, Intent var3) {
      MyLog.log("ActivityReminders.onActivityResult");
      if (var1 == 1) {
         if (var2 == -1) {
            MyLog.log("ActivityReminders.onActivityResult OK");
         }

         if (var2 == 0) {
            MyLog.log("ActivityReminders.onActivityResult CANCEL");
         }

         MyLog.log("ActivityReminders.onActivityResult loadFilters");
         this.a.a(this);
         this.b.notifyDataSetChanged();
         this.c.invalidate();
      }

   }

   public void onClick(View var1) {
      MyLog.log("ActivityReminders.onClick");
      if (var1 != null) {
         if (var1.getId() == R.id.b_reminders_select_add) {
            MyLog.log("ActivityReminders.onClick b_Add");
            this.startActivityForResult(new Intent(this, ActivityReminder.class), 1);
         }

      }
   }

   protected void onCreate(Bundle var1) {
      super.onCreate(var1);
      MyLog.log("ActivityReminders.onCreate");
      this.a = new MyReminders(this);
      MyLog.log("ActivityReminders.onCreate load");
      this.a.a(this);
      this.b = new MyRemindersArrayAdapter(this, this.a, this);
      this.setContentView(R.layout.activity_reminders);
      this.c = (ListView)this.findViewById(R.id.lv_reminders_select);
      this.c.setAdapter(this.b);
      this.c.setOnItemClickListener(new OnItemClickListener() {
         public void onItemClick(AdapterView var1, View var2, int var3, long var4) {
            StringBuilder var6 = new StringBuilder();
            var6.append("ActivityReminders.onItemClick ");
            var6.append(var2);
            MyLog.log(var6.toString());
         }
      });
      ((Button)this.findViewById(R.id.b_reminders_select_add)).setOnClickListener(this);
   }
}

File: app/src/main/java/com/dof100/morsenotifier/ActivityReminders.kt
package com.dof100.morsenotifier

import android.app.Activity
import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.Button
import android.widget.ListView
import com.dof100.morsenotifier.MyLog.log

class ActivityReminders constructor() : Activity(), View.OnClickListener,
  MyRemindersArrayAdapter.a {
  private var a: MyReminders? = null
  private var b: MyRemindersArrayAdapter? = null
  private var c: ListView? = null
  public override fun a(var1: MyReminder, var2: Int, var3: View) {
    log("ActivityReminders.onRowButtonClick")
    if (var3.getId() == R.id.b_delete) {
      log("ActivityReminders.onRowButtonClick b_delete")
      val var5: AlertDialog.Builder = AlertDialog.Builder(this)
      var5.setTitle(R.string.uninstall_free_title)
      var5.setMessage(R.string.title_activity_advanced)
      var5.setPositiveButton(R.string.action_yes, object : DialogInterface.OnClickListener {
        public override fun onClick(var1x: DialogInterface, var2: Int) {
          b!!.remove(var1)
          a!!.b(this@ActivityReminders)
          b!!.notifyDataSetChanged()
          MyJob.Companion.scheduleNextReminder(this@ActivityReminders as Context?)
          c!!.invalidate()
          var1x.dismiss()
        }
      })
      var5.setNegativeButton(R.string.action_no, object : DialogInterface.OnClickListener {
        public override fun onClick(var1: DialogInterface, var2: Int) {
          var1.dismiss()
        }
      })
      var5.create().show()
    } else {
      if (var3.getId() == R.id.b_edit) {
        log("ActivityReminders.onRowButtonClick b_edit")
        val var4: Intent = Intent(this, ActivityReminder::class.java)
        var4.putExtra("OBJECTINDEX", var2)
        this.startActivityForResult(var4, 1)
      }
    }
  }

  override fun onActivityResult(var1: Int, var2: Int, var3: Intent) {
    log("ActivityReminders.onActivityResult")
    if (var1 == 1) {
      if (var2 == -1) {
        log("ActivityReminders.onActivityResult OK")
      }
      if (var2 == 0) {
        log("ActivityReminders.onActivityResult CANCEL")
      }
      log("ActivityReminders.onActivityResult loadFilters")
      a!!.a(this)
      b!!.notifyDataSetChanged()
      c!!.invalidate()
    }
  }

  public override fun onClick(var1: View) {
    log("ActivityReminders.onClick")
    if (var1 != null) {
      if (var1.getId() == R.id.b_reminders_select_add) {
        log("ActivityReminders.onClick b_Add")
        this.startActivityForResult(Intent(this, ActivityReminder::class.java), 1)
      }
    }
  }

  override fun onCreate(var1: Bundle?) {
    super.onCreate(var1)
    log("ActivityReminders.onCreate")
    a = MyReminders(this)
    log("ActivityReminders.onCreate load")
    a!!.a(this)
    b = MyRemindersArrayAdapter(this, a, this)
    this.setContentView(R.layout.activity_reminders)
    c = findViewById<View>(R.id.lv_reminders_select) as ListView?
    c!!.setAdapter(b)
    c!!.setOnItemClickListener(object : OnItemClickListener {
      public override fun onItemClick(
        var1: AdapterView<*>?,
        var2: View,
        var3: Int,
        var4: Long
      ) {
        val var6: StringBuilder = StringBuilder()
        var6.append("ActivityReminders.onItemClick ")
        var6.append(var2)
        log(var6.toString())
      }
    })
    (findViewById<View>(R.id.b_reminders_select_add) as Button).setOnClickListener(this)
  }
}
